### PR TITLE
[Bug 20621] Refine indentation for some if variants

### PR DESCRIPTION
**Pull request by BerndN**
http://quality.livecode.com/show_bug.cgi?id=20621

on foo
   if foo() then
      if doo() then code else code -- <---
   else
      code
   end if
   -- comment
end foo

on foo
   if foo() then
      if doo() then code -- <---
   else
      code
   end if
end if
-- comment
end foo

### CODE REVIEW

**Comment by BerndN** [comment]
no bugfix note since there is already one for that bug number

File: Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript [hunk 1]
```diff
@@ -431,10 +431,16 @@ private function textFormatIndentLineRemoves pPreviousLine, pLine
...
       else
       
-         if "then" is among the words of pPreviousLine and token -1 of pPreviousLine is not "then" and token 1 of pPreviousLine is "if" and token - 1 of pLine is "else" then
+         if "then" is among the words of pPreviousLine and token -1 of pPreviousLine is not "then" \
+               and "else" is among the tokens of pPreviousLine and token 1 of pPreviousLine is "if" and token - 1 of pLine is "else" then
             return item 1 of sTextFormatKeywordMap["else"]
          end if
          
+         if "then" is among the words of pPreviousLine and token -1 of pPreviousLine is not "then" \
+               and token 1 of pPreviousLine is "if" and token - 1 of pLine is "else" then
```
Comment (montegoulding) [inline Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: @BerndN the space between `-` & `1` here gives me the hebegeebes... not sure if it does what we expect either... `token - 1 of pLine`. Actually same above.

Comment (BerndN) [reply Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: I could change it to "last token" or "token -1" if I knew how. Do I add a pull-request to this current one with the changes concerning the same lines as this one? I am using the web interface.

Comment (montegoulding) [reply Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: If you go to your fork you can choose the branch `patch-4` and then edit the file

Comment (BerndN) [reply Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript]: thanks, that worked, no more hebegeebees :)

**Comment by montegoulding** [comment]
@livecode-vulcan review ok d45b41c

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok d45b41c86c9a952a228289f74b7fc4003c3bd956

**Pull request merged**